code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -134,22 +134,21 @@
 	return ..()
 
 /obj/machinery/grill/proc/finish_grill()
+	if(grill_time >= 30)
+		SEND_SIGNAL(grilled_item, COMSIG_GRILL_FOOD)
 	switch(grill_time) //no 0-20 to prevent spam
 		if(20 to 30)
 			grilled_item.name = "lightly-grilled [grilled_item.name]"
 			grilled_item.desc = "[grilled_item.desc] It's been lightly grilled."
 		if(30 to 80)
 			grilled_item.name = "grilled [grilled_item.name]"
 			grilled_item.desc = "[grilled_item.desc] It's been grilled."
-			grilled_item.foodtype |= FRIED
 		if(80 to 100)
 			grilled_item.name = "heavily grilled [grilled_item.name]"
 			grilled_item.desc = "[grilled_item.desc] It's been heavily grilled."
-			grilled_item.foodtype |= FRIED
 		if(100 to INFINITY) //grill marks reach max alpha
 			grilled_item.name = "Powerfully Grilled [grilled_item.name]"
 			grilled_item.desc = "A [grilled_item.name]. Reminds you of your wife, wait, no, it's prettier!"
-			grilled_item.foodtype |= FRIED
```

Comment:
so uh
Only items with the edible component can get to this point, right?
So why not make everything about the "becoming grilled" reaction a part of the edible component? could move the name and desc setting there, alongside the switch. That way you wouldn't need a grill time check around the send signal too